src/dali/plugin/input/csv/binance.py

```diff
@@ -0,0 +1,219 @@
+from dali.configuration import Keyword
+from dali.in_transaction import InTransaction
+from dali.intra_transaction import IntraTransaction
+from dali.out_transaction import OutTransaction
+
+# transaction types
+_CRYPTO_DEPOSIT = "Crypto Deposit"
+_SELL = "Sell"
+_BUY = "Buy"
+_CRYPTO_WITHDRAWAL = "Crypto Withdrawal"
+_USD_DEPOSIT = "USD Deposit"
+_REFERRAL_COMMISSION = "Referral Commission"
+_STAKING_REWARDS = "Staking Rewards"
+_OTHERS = "Others"
+
+
+class InputPlugin(AbstractInputPlugin):
+
+    __BINANCE: str = "Binance.us"
+
+    __TIMESTAMP_INDEX = 1
+    __TRANSACTION_CATEGORY_INDEX = 2
+    __TRANSACTION_TYPE_INDEX = 3
+    __TRANSACTION_ID_INDEX = 5
+    __CURRENCY_INDEX = 6
+    __BASE_ASSET_INDEX = 9
+    __BASE_ASSET_AMOUNT_INDEX = 10
+    __BASE_ASSET_AMOUNT_SPOT_PRICE_INDEX = 11
+
+    # amounts in binance tax CSVs are not comma separated and are nicely formatting floating point values
+    __PRIMARY_ASSET_AMOUNT = 7
+    __PRIMARY_ASSET_SPOT_PRICE = 8
+
+    __DELIMITER = ","
+
+    def __init__(
+        self,
+        account_holder: str,
+        transaction_csv_file: str,
+        native_fiat: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder=account_holder, native_fiat=native_fiat)
+        self.__transaction_csv_file: str = transaction_csv_file
+        self.__logger: logging.Logger = create_logger(f"{self.__BINANCE}/{self.account_holder}")
+
+    def load(self) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        with open(self.__transaction_csv_file, encoding="utf-8") as transaction_csv_file:
+            # read CSV with header and skip first row
+            lines = reader(transaction_csv_file)
+
+            # Skip header line
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                transaction_id: str = line[self.__TRANSACTION_ID_INDEX].strip()
+                transaction_type: str = line[self.__TRANSACTION_TYPE_INDEX].strip()
+                category = line[self.__TRANSACTION_CATEGORY_INDEX].strip()
+                currency: str = line[self.__CURRENCY_INDEX].strip()
+
+                # there is no timezone information in the CSV, so we assume UTC
+                timestamp_with_timezone = f"{line[self.__TIMESTAMP_INDEX].strip()} -00:00"
+
+                common_params = {
+                    "plugin": self.__BINANCE,
+                    "unique_id": transaction_id,
+                    "raw_data": raw_data,
+                    "timestamp": timestamp_with_timezone,
+                    "notes": f"{category} - {transaction_type}",
+                }
+
+                # in binance, you can pay fees with lots of different currencies. Since dali does not allow us to specify a fee currency, we use the normalized USD fee amount
```

Comment:
Fiat and crypto fees (both in the same currency or in different one) can be represented in DaLI and RP2. Check this FAQ on the subject: https://github.com/eprbell/rp2/blob/main/docs/user_faq.md#how-to-represent-fiat-vs-crypto-transaction-fees

Reply:
Got it, very helpful!